From 9c3cb1b63a0cab0f53f596506ae09d9c545711fa Mon Sep 17 00:00:00 2001
From: Tzvetomir Stoyanov <tz.stoyanov@gmail.com>
Date: Tue, 11 Nov 2025 05:12:30 +0200
Subject: [PATCH 4/5] Allow redefine of FOTA compile time options

Usually FOTA is used as an external library to a project. There are
a few compile time options, that the user of the library can tweak
depending on the use case. Currently these can be changed only by
modifying the FOTA CMake file. Allowing user to define these options in
its CMake files simplifies the FOTA library integration.
---
 CMakeLists.txt | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/libs/pico_fota_bootloader/CMakeLists.txt b/libs/pico_fota_bootloader/CMakeLists.txt
index ff6f93c..979a06e 100644
--- a/libs/pico_fota_bootloader/CMakeLists.txt
+++ b/libs/pico_fota_bootloader/CMakeLists.txt
@@ -30,12 +30,21 @@ endif()
 ########################################
 # CMake options
 ########################################
-option(PFB_WITH_BOOTLOADER_LOGS "Enables logging messages from the bootloader using stdio" ON)
-option(PFB_REDIRECT_BOOTLOADER_LOGS_TO_UART "Redirects bootloader's logs from USB to UART" OFF)
-option(PFB_WITH_IMAGE_ENCRYPTION "Enables image encryption using AES ECB algorithm" ON)
-option(PFB_AES_KEY "AES key used for image encryption and decryption")
-option(PFB_WITH_SHA256_HASHING "Enables image SHA256 appending and checking" ON)
-
+if ( NOT DEFINED PFB_WITH_BOOTLOADER_LOGS )
+    option(PFB_WITH_BOOTLOADER_LOGS "Enables logging messages from the bootloader using stdio" ON)
+endif()
+if ( NOT DEFINED PFB_REDIRECT_BOOTLOADER_LOGS_TO_UART )
+    option(PFB_REDIRECT_BOOTLOADER_LOGS_TO_UART "Redirects bootloader's logs from USB to UART" OFF)
+endif()
+if ( NOT DEFINED PFB_WITH_IMAGE_ENCRYPTION )
+    option(PFB_WITH_IMAGE_ENCRYPTION "Enables image encryption using AES ECB algorithm" ON)
+endif()
+if ( NOT DEFINED PFB_AES_KEY )
+    option(PFB_AES_KEY "AES key used for image encryption and decryption")
+endif()
+if ( NOT DEFINED PFB_WITH_SHA256_HASHING )
+    option(PFB_WITH_SHA256_HASHING "Enables image SHA256 appending and checking" ON)
+endif()
 if ( NOT PFB_RESERVED_FILESYSTEM_SIZE_KB )
     set(PFB_RESERVED_FILESYSTEM_SIZE_KB 0)
 endif()
-- 
2.51.0

